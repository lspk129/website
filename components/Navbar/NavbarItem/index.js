import styled from 'styled-components';

const NavbarItem = styled.a`
  color: ${props => props.theme.text};
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 20px;
  cursor: pointer;
  display: inline-block;
  &:after {
    display: block;
    content: '';
    border-bottom: solid 3px ${props => props.theme.text};  
    transform: scaleX(0);  
    transition: transform 250ms ease-in-out;
  }
  &:hover:after {
    transform: scaleX(1);
  }

`;

export default NavbarItem;
