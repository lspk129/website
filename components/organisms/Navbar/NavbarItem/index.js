import styled from 'styled-components';

const NavbarItem = styled.div`
  color: ${props => props.theme.text};
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 16px;
  cursor: pointer;
  &:after {
    display: grid;
    content: '';
    border-bottom: solid 2px ${props => props.theme.text};  
    transform: scaleX(0);  
    transition: transform 250ms ease-in-out;
  }
  &:hover:after {
    transform: scaleX(1);
  }

`;

export default NavbarItem;
