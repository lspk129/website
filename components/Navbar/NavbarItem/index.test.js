import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NavbarItem from './';

describe('NavbarItem', () => {
  const shallowWrapper = shallow(<NavbarItem />);

  it('renders correctly', () => {
    const wrapper = renderer.create(<NavbarItem />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders div element', () => {
    expect(shallowWrapper).toBePresent();
    expect(shallowWrapper).toHaveTagName('div');
  });
});
