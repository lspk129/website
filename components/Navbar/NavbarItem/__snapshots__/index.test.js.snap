// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarItem renders correctly 1`] = `
<div
  className="sc-bdVaJa hgwNKU"
/>
`;
